R/package-auth.R

```diff
@@ -20,29 +22,36 @@
 #' - `use_databrickscfg`: Boolean (default: `FALSE`), determines if credentials
 #' are fetched from profile of `.databrickscfg` or `.Renviron`
 #' - `db_profile`: String (default: `NULL`), determines profile used.
+#' When the profile `workbench` is set then `.databrickscfg` will
+#' automatically be searched to enable the use of Posit Workbench managed credentials.
 #'
 #' See vignette on authentication for more details.
 #'
 #' @family Databricks Authentication Helpers
 #'
 #' @return workspace URL
 #' @export
-db_host <- function(id = NULL, prefix = NULL, profile = getOption("db_profile", NULL)) {
+db_host <- function(id = NULL, prefix = NULL, profile = default_config_profile()) {
+  if (is.null(id) && is.null(prefix)) {
 
-  # if option `use_databrickscfg` is `TRUE` then fetch the associated env.
-  # env is specified via `db_env` option, if missing use default.
-  # this behaviour can only be changed via setting of config
-  if (getOption("use_databrickscfg", FALSE)) {
-    host <- read_databrickscfg(key = "host", profile = profile)
-    return(host)
-  }
+    use_databricks_cfg <- getOption("use_databrickscfg", FALSE)
+    if (profile == "workbench") {
+      use_databricks_cfg <- TRUE
+    }
```

Comment:
My worry is that `workbench` is not specific enough and its foreseeable that people might name their profiles 'workbench' otherwise. E.g. I know of at least one customer that would hit this.

Reply:
I can see that. @atheriel is there a better way to tell if running on Workbench? Maybe an env variable that only exists when in a PWB session?

Reply:
We could use what the `odbc` package code uses: https://github.com/r-dbi/odbc/blob/ab9421fa7868a383696b35515127d2a0b87d5d42/R/driver-databricks.R#L256C6-L256C57

Reply:
I think that checking if the config path contains `"posit-workbench"` would work, yes.